demisto_sdk/commands/validate/tests/IF_validators_test.py

```diff
@@ -0,0 +1,332 @@
+from typing import List
+
+import pytest
+
+from demisto_sdk.commands.content_graph.objects.incident_field import IncidentField
+from demisto_sdk.commands.validate.tests.test_tools import create_incident_field_object
+from demisto_sdk.commands.validate.validators.IF_validators.IF100_is_valid_name_and_cli_name import (
+    IsValidNameAndCliNameValidator,
+)
+from demisto_sdk.commands.validate.validators.IF_validators.IF101_is_valid_content_field import (
+    IsValidContentFieldValidator,
+)
+from demisto_sdk.commands.validate.validators.IF_validators.IF102_is_valid_system_flag import (
+    IsValidSystemFlagValidator,
+)
+from demisto_sdk.commands.validate.validators.IF_validators.IF103_is_valid_field_type import (
+    FIELD_TYPES,
+    IsValidFieldTypeValidator,
+)
+from demisto_sdk.commands.validate.validators.IF_validators.IF104_is_valid_group_field import (
+    INCIDENT_FIELD_GROUP,
+    IsValidGroupFieldValidator,
+)
+from demisto_sdk.commands.validate.validators.IF_validators.IF105_is_cli_name_field_alphanumeric import (
+    IsCliNameFieldAlphanumericValidator,
+)
+from demisto_sdk.commands.validate.validators.IF_validators.IF106_is_cli_name_reserved_word import (
+    IsCliNameReservedWordValidator,
+)
+
+
+@pytest.mark.parametrize(
+    "content_items, expected_msg",
+    [
+        pytest.param(
+            [create_incident_field_object(["name", "cliName"], ["case 1", "case1"])],
+            "The following words cannot be used as a name: case.",
+            id="One IncidentField with bad word in field `name`",
+        ),
+        pytest.param(
+            [
+                create_incident_field_object(
+                    ["name", "cliName"], ["case incident 1", "caseincident1"]
+                ),
+            ],
+            "The following words cannot be used as a name: case, incident.",
+            id="IncidentField with two bad words in field `name`",
+        ),
+    ],
+)
+def test_IsValidNameAndCliNameValidator_not_valid(
+    content_items: List[IncidentField],
+    expected_msg: List[str],
+):
+    """
+    Given:
+        - IncidentFields content items
+    When:
+        - run is_valid method
+    Then:
+        Case 1:
+            - Ensure the error message is as expected
+            - Ensure number of failure is as expected
+        Case 2:
+            - Ensure the error message is as expected with the bad words list
+    """
+    results = IsValidNameAndCliNameValidator().is_valid(content_items=content_items)
+    assert results[0].message == expected_msg
```

Comment:
If the validation doesn't fail, the test will fail on a `KeyError`, and it'll be a bit difficult to understand why.
Please add `assert results` to all tests where we need to know the validation detected a violation